Message-ID: <45FB2663.1010100@kitware.com>
Date: Fri, 16 Mar 2007 19:21:07 -0400
From: Brad King <brad.king@kitware.com>
To: Martin Costabel <costabel@wanadoo.fr>
Subject: Re: [vtkusers] Interacting with VTK under Mac OS X
Cc: vtkusers@vtk.org

This is a multi-part message in MIME format.
--------------020804060404070505000209
Content-Type: text/plain; charset=ISO-8859-1
Content-Transfer-Encoding: 7bit

Martin Costabel wrote:
> The problem is that, as others have answered already, that nobody seems
> to know how to get this to work. I have read the documentation and the
> sources many times forwards and backwards, and I have literally spent
> several days in vain trying to get CMake for VTK to produce
> install_names like /sw/lib/vtk/libvtkCommon.5.0.dylib.
> 
> If the result of the present discussion is a working example for VTK
> libraries, I would be very glad.

The reason it doesn't work in VTK 5.0 is because that was released
before CMake 2.4 so it does a big hack to prevent installation with any
kind of rpath enabled.  Also the default behavior for the release branch
is to have rpaths off.  Apply the attached patch to re-enable
installation with rpaths.  Then run

cmake ../VTK-5.0 \
  -DCMAKE_INSTALL_PREFIX=/sw \
  -DBUILD_SHARED_LIBS=ON \
  -DCMAKE_INSTALL_NAME_DIR:STRING=/sw/lib \
  -DVTK_USE_RPATH=ON

This should get it working for most of the libraries.  Then instead of
using -DCMAKE_INSTALL_NAME_DIR:STRING=/sw/lib you can add

SET(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib)

to the source tree.  You may also need to use SET_TARGET_PROPERTIES to
tweak the setting for a few targets, but probably not.

-Brad

--------------020804060404070505000209
Content-Type: text/x-diff;
 name="vtk-5.0-install.patch"
Content-Transfer-Encoding: 7bit
Content-Disposition: inline;
 filename="vtk-5.0-install.patch"

Index: CMakeLists.txt
===================================================================
RCS file: /cvsroot/VTK/VTK/CMakeLists.txt,v
retrieving revision 1.341.2.19
diff -c -3 -p -r1.341.2.19 CMakeLists.txt
*** CMakeLists.txt	6 Feb 2007 18:51:59 -0000	1.341.2.19
--- CMakeLists.txt	16 Mar 2007 23:14:52 -0000
*************** INCLUDE(${CMAKE_ROOT}/Modules/CMakeExpor
*** 40,54 ****
  # Choose static or shared libraries.
  INCLUDE(${VTK_SOURCE_DIR}/CMake/vtkSelectSharedLibraries.cmake)
  
- # Disable the install targets if using the rpath feature.
- IF(NOT WIN32)
-   IF(VTK_USE_RPATH)
-     SET(VTK_INSTALL_NO_RUNTIME 1)
-     SET(VTK_INSTALL_NO_DEVELOPMENT 1)
-     SET(VTK_INSTALL_NO_DOCUMENTATION 1)
-   ENDIF(VTK_USE_RPATH)
- ENDIF(NOT WIN32)
- 
  #-----------------------------------------------------------------------------
  # Output directories.
  IF(NOT LIBRARY_OUTPUT_PATH)
--- 40,45 ----
Index: CMake/vtkSelectSharedLibraries.cmake
===================================================================
RCS file: /cvsroot/VTK/VTK/CMake/vtkSelectSharedLibraries.cmake,v
retrieving revision 1.1
diff -c -3 -p -r1.1 vtkSelectSharedLibraries.cmake
*** CMake/vtkSelectSharedLibraries.cmake	11 Aug 2005 19:49:09 -0000	1.1
--- CMake/vtkSelectSharedLibraries.cmake	16 Mar 2007 23:14:52 -0000
*************** IF(NOT VTK_SHARED_LIBRARIES_SELECTED)
*** 48,64 ****
      IF(VTK_USE_RPATH)
        # We will use rpath support.  Tell CMake not to skip it.
        SET(CMAKE_SKIP_RPATH 0 CACHE INTERNAL "Whether to build with rpath." FORCE)
- 
-       # If someone is trying to install do not do an entire build with
-       # the wrong rpath feature setting just to report failed
-       # installation.
-       SET(CMAKE_SKIP_INSTALL_ALL_DEPENDENCY 1)
- 
-       # Add a dummy target and attach an install rule that will always fail
-       # and produce a message explaining why installation is disabled.
-       ADD_CUSTOM_TARGET(vtk_install_disabled)
-       SET_TARGET_PROPERTIES(vtk_install_disabled PROPERTIES
-         PRE_INSTALL_SCRIPT ${VTK_SOURCE_DIR}/CMake/InstallDisabled.cmake)
      ELSE(VTK_USE_RPATH)
        # We will not use rpath support.  Tell CMake to skip it.
        SET(CMAKE_SKIP_RPATH 1 CACHE INTERNAL "Whether to build with rpath." FORCE)
--- 48,53 ----

--------------020804060404070505000209
Content-Type: text/plain; charset="us-ascii"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Content-Disposition: inline

_______________________________________________
This is the private VTK discussion list. 
Please keep messages on-topic. Check the FAQ at: http://www.vtk.org/Wiki/VTK_FAQ
Follow this link to subscribe/unsubscribe:
http://www.vtk.org/mailman/listinfo/vtkusers

--------------020804060404070505000209--



